components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt

```diff
@@ -44,6 +46,26 @@ open class PlacesHistoryStorage(
     context: Context,
     crashReporter: CrashReporting? = null
 ) : PlacesStorage(context, crashReporter), HistoryStorage, HistoryMetadataStorage, SyncableStore {
+    /**
+     * Backing property for the reader used in all queries in this this class.
+     * Allows to easily change the reader in tests. Otherwise can be ignored.
+     * This allows for better encapsulation. It is `private` while [reader] is `internal`.
+     */
+    @VisibleForTesting
+    @SuppressWarnings("VariableNaming")
+    internal var _reader: PlacesReaderConnection? = null
+
+    /**
+     * Allows to read/write history data without any impact on other storage queries from other features.
+     */
+    override val reader: PlacesReaderConnection
+        // Allow the reader to be injected from tests otherwise use a quick unsafe singleton in this class.
+        get() = if (_reader != null) {
+            _reader!!
+        } else {
+            _reader = places.newReader()
+            _reader!!
+        }
```

Comment:
Not the prettiest but this 
- allows creating only one reader collection (vs just overriding get() without the backing `_reader`) and 
- allows to postpone the `places.newReader()` by using a getter for this property to after the construction of this and parent is finished and in that support the [`MockingPlacesHistoryStorage`](https://github.com/mozilla-mobile/android-components/blob/fddd02dce9f5c92906e071310a0b6118e130e1be/components/browser/storage-sync/src/test/java/mozilla/components/browser/storage/sync/PlacesHistoryStorageTest.kt#L590) from tests which overrides `places` but after the `places.newReader()`call and would otherwise throw a NPE.

Reply:
Not applicable anymore.